cryostatio/cryostat-operator
# fix(tls): update existing certificates

Uses `controllerutil.CreateOrUpdate` on certificates/issuers/secrets created in certmanager.go. I've also added some additional tests for this code.

To test:
```
make deploy IMG=quay.io/ebaron/cryostat-operator:before-352
make create_cryostat_cr
<wait for cryostat-sample-grafana-tls secret to be created>
kubectl patch deploy cryostat-operator-controller-manager -p \
'{"spec":{ "template": {"spec": {"containers": [{"name": "manager", "image": "quay.io/ebaron/cryostat-operator:after-352-update"}]}}}}'
<redeployed Cryostat container logs should show no CertificateException>
```

`quay.io/ebaron/cryostat-operator:before-352` is a build before #352 
`quay.io/ebaron/cryostat-operator:after-352-update` is a build of this PR, after #352 

Fixes: #359
